vernac/declare.ml

```diff
@@ -1765,7 +1793,18 @@ let prepare_proof ?(warn_incomplete=true) ~unsafe_typ { proof } =
     | Some p ->
       Vars.universes_of_constr p, p
     | None ->
-      CErrors.user_err Pp.(str "Some unresolved existential variables remain")
+      let has_given_up =
+        let exception Found in
+        let rec aux c =
+          let () = match EConstr.kind evd c with
```

Comment:
This one check looks very hackish, couldn't we have a dedicated API for that?

Reply:
It's not hack it's adhoc, which is fine.

Reply:
Hack (< \*a(d)h(o)c)?

I'm not complaining strongly but this is still ugly.